### PR TITLE
Continued sessions

### PR DESCRIPTION
**Pull request by galactoise**
Changes conversationHistory to be Deque instead of a List, and starts a utility for searching conversation history

### CODE REVIEW

**Comment by galactoise** [comment]
Also, depends on PR #12

File: service/src/main/java/com/derpgroup/derpwizard/manager/AbstractManager.java [hunk 1]
```diff
@@ -43,10 +36,12 @@
  */
 public abstract class AbstractManager {
   
-  protected Module mapperModule;
+  //protected Module mapperModule;
+  protected ObjectMapper mapper;
 
   public AbstractManager(){
-    mapperModule = new SimpleModule();
+  //  mapperModule = new SimpleModule();
```
Comment (davidwphillips) [inline service/src/main/java/com/derpgroup/derpwizard/manager/AbstractManager.java]: Wipe this and the commented declaration of it above. We're not using it anymore

File: service/src/main/java/com/derpgroup/derpwizard/manager/AbstractManager.java [hunk 2]
```diff
@@ -58,7 +53,8 @@ public AbstractManager(){
    *          The document builder to append messages to, not null
    */
   public void handleRequest(@NonNull VoiceInput voiceInput, @NonNull SsmlDocumentBuilder builder) {
-    appendToConversationHistory(voiceInput);
+    CommonMetadata metadata = voiceInput.getMetadata();
+    ConversationHistoryUtils.appendToConversationHistory(voiceInput, metadata, mapper);
```
Comment (davidwphillips) [inline service/src/main/java/com/derpgroup/derpwizard/manager/AbstractManager.java]: Why are we extracting CommonMetadata from VoiceInput if we're turning around and passing both into this function right here.

File: voice-interface/src/main/java/com/derpgroup/derpwizard/voice/util/ConversationHistoryUtils.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      if(entry.getMessageSubject() != null && !metaSubjects.contains(entry.getMessageSubject())){
+        return entry;
+      }
+    }
+    
+    return null; //Should this return null?
+  }
+  
+  //Should this operate on the object directly, or return something?
+  public static void appendToConversationHistory(@NonNull VoiceInput voiceInput, CommonMetadata metadata, ObjectMapper mapper) {
```
Comment (davidwphillips) [inline voice-interface/src/main/java/com/derpgroup/derpwizard/voice/util/ConversationHistoryUtils.java]: This is the other side of my other comment. CommonMetadata is an internal variable inside VoiceInput. So why are we passing both here?

**Comment by galactoise** [comment]
@davidwphillips What do you think of this approach to the mapper?  It still doesn't address the metadata vs. voiceinput question, but it at least makes the rest of the request less awkward

**Pull request merged**